WooCommerce/Classes/ViewRelated/Products/ProductLoaderViewController.swift

```diff
@@ -0,0 +1,246 @@
+        configureDismissButton()
+        loadProduct()
+    }
+}
+
+
+// MARK: - Configuration
+//
+private extension ProductLoaderViewController {
+
+    /// Setup: Navigation Title
+    ///
+    func configureNavigationTitle() {
+        title = NSLocalizedString("Loading Product", comment: "Displayed when an Product is being retrieved")
+    }
+
+    /// Setup: Main View
+    ///
+    func configureMainView() {
+        view.backgroundColor = StyleManager.tableViewBackgroundColor
+        view.addSubview(activityIndicator)
+        view.pinSubviewAtCenter(activityIndicator)
+    }
+
+    /// Setup: Spinner
+    ///
+    func configureSpinner() {
+        activityIndicator.hidesWhenStopped = true
+        activityIndicator.translatesAutoresizingMaskIntoConstraints = false
+    }
+
+    /// Setup: Dismiss Button
+    ///
+    func configureDismissButton() {
+        let dismissButtonTitle = NSLocalizedString("Dismiss", comment: "Product details screen - button title for closing the view")
+        let leftBarButton = UIBarButtonItem(title: dismissButtonTitle,
+                                            style: .plain,
+                                            target: self,
+                                            action: #selector(dismissButtonTapped))
+        leftBarButton.tintColor = .white
+        navigationItem.setLeftBarButton(leftBarButton, animated: false)
+    }
+}
+
+
+// MARK: - Actions
+//
+private extension ProductLoaderViewController {
+
+    /// Loads (and displays) the specified Product.
+    ///
+    func loadProduct() {
+        let action = ProductAction.retrieveProduct(siteID: siteID, productID: productID) { [weak self] (product, error) in
+            guard let `self` = self else {
+                return
+            }
+
+            guard let product = product else {
+                DDLogError("⛔️ Error loading Product for siteID: \(self.siteID) productID:\(self.productID) error:\(error.debugDescription)")
+                self.state = .failure
+                return
+            }
+
+            self.state = .success(product: product)
+        }
+
+        state = .loading
+        StoresManager.shared.dispatch(action)
+    }
+
+    @objc func dismissButtonTapped() {
+        dismiss(animated: true, completion: nil)
+    }
+}
+
+
+// MARK: - Overlays
+//
+private extension ProductLoaderViewController {
+
+    /// Starts the Spinner
+    ///
+    func startSpinner() {
+        activityIndicator.startAnimating()
+    }
+
+    /// Stops the Spinner
+    ///
+    func stopSpinner() {
+        activityIndicator.stopAnimating()
+    }
+
+    /// Displays the Loading Overlay.
+    ///
+    func displayFailureOverlay() {
+        let overlayView: OverlayMessageView = OverlayMessageView.instantiateFromNib()
+        overlayView.messageImage = .waitingForCustomersImage
+        overlayView.messageText = NSLocalizedString("The Product couldn't be loaded.", comment: "Fetching a product failed")
```

Comment:
Maybe re-phrase to "This product couldn't be loaded." ? 

Or lowercase the word "product" in the original message since it's not a proper noun.

Reply:
WFM — I will update!